documentation/directory-structure.md

```diff
@@ -0,0 +1,247 @@
+    - [Versioning services in a service group](#versioning-services-in-a-service-group)
+  - [Deprecated structure and hand-written OpenAPI specs](#deprecated-structure-and-hand-written-openapi-specs)
+# `specification` directory structure
+
+
+This article describes the directory structure / folder layout of the `specification` folder.
+You may be also interested in following:
+- [Specification index]
+- [Resource Provider list]
+
+> [!IMPORTANT]
+> The structure described in this article is strictly enforced. There exist some exceptions for historical reasons.
+> These exceptions are not allowed going forward.
+
+## `specification` folder
+
+The `specification` folder is the root folder for all service specifications.
+
+Each child of the `specification` folder corresponds to a `service` specification for given Azure team. Here we denote such folder as `<azureTeam>`.
+In advanced cases for big teams the `<azureTeam>` folder can host multiple services, known as `service group`.
+Read [the relevant section](#advanced-scenario-service-group) for details.
+
+Given `<azureTeam>` has following structure:
+
+- `<azureTeam>/<typeSpecFolder>` (multiple folders)
+- `<azureTeam>/resource-manager`
+- `<azureTeam>/data-plane`
+
+The `<azureTeam>/<typeSpecFolder>` folders contain the TypeSpec specification for the given `service` or `service group`.
+You can find details on the name and contents of these folders in [TypeSpec directory structure].
+You can learn more about TypeSpec at [aka.ms/azsdk/typespec] and [aka.ms/typespec].
+
+## `resource-manager` and `data-plane` folders
+
+The `<azureTeam>/resource-manager` contains the ARM OpenAPI specifications emitted from TypeSpec in `<azureTeam>/<typeSpecFolder>`.
+
+The `resource-manager` folder has exactly one child folder whose name matches the **Resource Provider (RP) Namespace** (`<RPNS>`), 
+such as `Microsoft.Automation` (full list of namespaces is [here][Resource Provider list]).
+There is 1 to 1 correspondence between an RP and an RP namespace.
+There must be **exactly one** RP namespace folder under given `resource-manager` folder.
+We denote such folder as `resource-manager/<RPNS>`.
+
+The `<azureTeam>/data-plane` contains the data-plane OpenAPI specifications emitted from TypeSpec in `<typeSpecFolder>`. 
+The `data-plane` folder has no child `<RPNS>` folder. However, it can have a set of `<dataPlaneSubfolder>` folders.
+
+## AutoRest configuration `README.md` files
+
+Both the `<azureTeam>/resource-manager` and `<azureTeam>/data-plane` folders must contain an AutoRest configuration file, `README.md`.
+Learn more about this file at [aka.ms/azsdk/autorest].
+
+<!-- This paragraph about README contents is a bit vague. We should improve it, eventually. -->
+
+Each `README.md` describes a single `service` and is used as an SDK package and documentation for each version of the service.
+Inside the `README.md` file there are lists of paths to OpenAPI spec `.json` files making up given service version.
+
```

Comment:
I think we should describe specific requirements on the README.md here. For example, it should contain a "default" tag that refers to "current" version of the service (the ref docs depend on this).

Reply:
To avoid duplication I wanted to keep this section lightweight. But perhaps we should add it to https://aka.ms/azsdk/autorest.

Note this doc mentions:
> Both the <azureTeam>/resource-manager and <azureTeam>/data-plane folders must contain an AutoRest configuration file, README.md. Learn more about this file at [aka.ms/azsdk/autorest](https://aka.ms/azsdk/autorest).